tools/run_tests/xds_k8s_test_driver/bin/cleanup.sh

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
+# Copyright 2021 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eo pipefail
+
+display_usage() {
+  cat <<EOF >/dev/stderr
+Performs full TD and K8S resource cleanup
+
+USAGE: $0 [--secure] [arguments]
+   --secure: Perform TD resource cleanup specific for PSM Security setup
+   arguments ...: additional arguments passed to ./run.sh
+
+ENVIRONMENT:
+   XDS_K8S_CONFIG: file path to the config flagfile, relative to
+                   driver root folder. Default: config/local-dev.cfg
+                   Will be appended as --flagfile="config_absolute_path" argument
+   XDS_K8S_DRIVER_VENV_DIR: the path to python virtual environment directory
+                            Default: $XDS_K8S_DRIVER_DIR/venv
+EXAMPLES:
+$0
+$0 --secure
+XDS_K8S_CONFIG=./path-to-flagfile.cfg $0 --namespace=override-namespace
+EOF
+  exit 1
+}
+
+if [[ "$1" == "-h" || "$1" == "--help" ]]; then
+  display_usage
+fi
+
+readonly SCRIPT_DIR="$( cd -- "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
+readonly XDS_K8S_DRIVER_DIR="${SCRIPT_DIR}/.."
+
+cd "${XDS_K8S_DRIVER_DIR}"
+
+if [[ "$1" == "--secure" ]]; then
```

Comment:
Is `--secure` necessary here? We could try to delete secure resources for non-secure setups. According to the code, I might be wrong, it seems a pure addition of policy resources? Less argument -> easier to use.

Reply:
There used to be a difference in what type of healtchecks are used for secure and non-secure setups. I agree with the simplification tho. Let me see what I can do.